csv/result.go

```diff
@@ -350,7 +350,20 @@ func readMetadata(r *csv.Reader, c ResultDecoderConfig, extraLine []string) (tab
 					return tableMetadata{}, fmt.Errorf("missing expected annotation default")
 				}
 			}
-			// Skip extra annotation
+			if !strings.HasPrefix(line[annotationIdx], commentPrefix) {
```

Comment:
I think it would be better to change the conditional above to:

```
if !strings.HasPrefix(line[annotationIdx], commentPrefix) {
}
```

And then keep everything else the same. I like the change in error message, but it needs to be consistent with the rest of the error messages and so I'd prefer to keep the fix version of this change to as few error message changes as possible. We could do a follow up change to improve the error message afterwards.